ext/extconf.rb

```diff
@@ -1,63 +1,85 @@
+require 'digest'
+require 'logger'
 require 'mkmf'
 require 'fileutils'
 require 'open-uri'
 require 'zlib'
 require 'rubygems/package'
+require 'yaml'
 require File.expand_path('../../lib/appsignal/version.rb', __FILE__)
 
-HOST            = 'd135dj0rjqvssy.cloudfront.net'
-SUPPORTED_ARCHS = %w(x86_64-linux x86_64-darwin)
-EXT_PATH        = File.expand_path('..', __FILE__)
-LIB_EXTENSION   = case Gem::Platform.local.os
-                  when /darwin/
-                    'dylib'
-                  else
-                    'so'
-                  end
+EXT_PATH     = File.expand_path('..', __FILE__)
+AGENT_CONFIG = YAML.load(File.read(File.join(EXT_PATH, 'agent.yml')))
+ARCH         = "#{Gem::Platform.local.cpu}-#{Gem::Platform.local.os}"
 
 def ext_path(path)
   File.join(EXT_PATH, path)
 end
 
+def logger
+  @logger ||= Logger.new(File.join(EXT_PATH, 'install.log'))
+end
+
 def installation_failed(reason)
-  puts "Installation failed: #{reason}"
+  logger.error "Installation failed: #{reason}"
   File.open(File.join(EXT_PATH, 'Makefile'), 'w') do |file|
       file.write "default:\nclean:\ninstall:"
   end
 end
 
-arch = "#{Gem::Platform.local.cpu}-#{Gem::Platform.local.os}"
-
-if SUPPORTED_ARCHS.include?(arch)
-  begin
-    if !File.exists?(ext_path('appsignal-agent')) || !File.exists?(ext_path("libappsignal.#{LIB_EXTENSION}"))
-      archive_url = "https://#{HOST}/#{Appsignal::AGENT_VERSION}/appsignal-agent-#{arch}.tar.gz"
-      archive     = open(archive_url)
-      Gem::Package::TarReader.new(Zlib::GzipReader.open(archive)) do |tar|
-        tar.each do |entry|
-          if entry.file?
-            File.open(ext_path(entry.full_name), 'wb') do |f|
-              f.write(entry.read)
-            end
+def install
+  logger.info "Installing appsignal agent for Ruby #{RUBY_VERSION} on #{RUBY_PLATFORM}"
+
+  unless AGENT_CONFIG[:triples].keys.include?(ARCH)
+    installation_failed(
+      "AppSignal currently does not support your system architecture (#{ARCH})." \
+      "Please let us know at support@appsignal.com, we aim to support everything our customers run."
```

Comment:
This will run on either a development machine or a laptop, we don't have a way to detect how we can reliably send e-mail for everything. Also I'm not sure how people would feel about it.